rust/lance/src/index/create.rs

```diff
@@ -454,12 +454,21 @@ mod tests {
             "Should have multiple fragments for testing"
         );
 
-        // Test fragments() method with specific fragment IDs
-        let selected_fragments = vec![fragment_ids[0], fragment_ids[1]];
+        // Test fragments() method with specific fragment IDs and ensure duplicate/out-of-order fragments are handled properly
+        let selected_fragments = vec![
+            fragment_ids[2],
+            fragment_ids[1],
+            fragment_ids[0],
+            fragment_ids[1],
+        ];
+        let mut selected_fragments_expected = selected_fragments.clone();
```

Comment:
I think the expected should be spelled out as

```rust
vec![
            fragment_ids[0],
            fragment_ids[1],
            fragment_ids[2],
        ];
```

so that we are sure sort and dedupe are behaving correctly